### PR TITLE
ssh user@host implementation (draft)

### PR DESCRIPTION
**Pull request by ilmanzo**
approaching issue  #38

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,12 +2,15 @@ module github.com/dlespiau/footloose
 
 require (
 	github.com/ghodss/yaml v1.0.0
-	github.com/inconshreveable/mousetrap v1.0.0 // indirect
+	github.com/kr/pretty v0.1.0 // indirect
```
Comment (dlespiau) [inline go.mod]: It looks like you may have imported a dependency at some point. This happens to me with my IDE trying to be clever (with goimport) and start importing things form my GOTPATH. And then building the package modifies `go.mod` and `go.sum`.

This should be cleaned up, I don't believe we need any change in `footloose` dependencies for this change.

File: tests/t_old/test-create-delete-key [hunk 2]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (dlespiau) [inline tests/t_old/test-create-delete-key]: `t_old` seems like something you have been doing locally to test, it should be removed from this PR.

File: tests/test-ssh-remote-command.cmd [hunk 3]
```diff
@@ -1,4 +1,4 @@
-footloose config create --config %t.footloose --name %t --key %t-key
-footloose create --config %t.footloose
-%out footloose --config %t.footloose ssh node0 hostname
-footloose delete --config %t.footloose
+../footloose config create --config %t.footloose --name %t --key %t-key
```
Comment (dlespiau) [inline tests/test-ssh-remote-command.cmd]: the tests expert `footloose` to be in the path, don't think we need to change that, maye using `go install` would help here. Something like:

```
go install # in the root of this repository
export PATH=`go env GOPATH`/bin:$PATH
```

File: tests/test-ssh-remote-command.cmd [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
+../footloose create --config %t.footloose
+%out ../footloose --config %t.footloose ssh root@node0 whoami
```
Comment (dlespiau) [inline tests/test-ssh-remote-command.cmd]: Yes, that tests seems great to me to sure we do login as root. You've added it as a separate test, which is nice, but we shouldn't then change the `test-ssh-remote-command` test.

`test-ssh-remote-command.*` should be left untouched.

File: tests/test-ssh-user.cmd [hunk 5]
```diff
@@ -0,0 +1,4 @@
+../footloose config create --config %t.footloose --name %t --key %t-key
```
Comment (dlespiau) [inline tests/test-ssh-user.cmd]: the tests expert `footloose` to be in the path, don't think we need to change that and so we should remove the `../`. Maybe using `go install` would help here. Something like:

```
go install # in the root of this repository
export PATH=`go env GOPATH`/bin:$PATH
```

**Comment by ilmanzo** [comment]
I'm afraid the test with ssh node0 will ask the user password ... So user creation and ssh key configuration must be implemented before .

**Comment by dlespiau** [comment]
Ah, sorry, hadn't fully realised what we needed to do. I think all should be fine if we tune all the places where we ssh in from:

```
footloose --config %t.yaml ssh node0
```

to:

```
footloose --config %t.yaml ssh root@node0
```

That seems to include;
- `README.md`, telling people how to ssh as root in with that change
- `footloose/tests/test-docker-in-docker.cmd`
- `footloose/tests/test-ssh-remote-command.cmd`

**Pull request merged**